allennlp/models/model.py

```diff
@@ -293,7 +294,9 @@ def _load(
         # stored in our weights.  We don't need any pretrained weight file anymore, and we don't
         # want the code to look for it, so we remove it from the parameters here.
         remove_pretrained_embedding_params(model_params)
-        model = Model.from_params(vocab=vocab, params=model_params)
+        model = Model.from_params(
+            vocab=vocab, params=model_params, postpone_cleanup=postpone_cleanup
```

Comment:
Why does this need to be passed to `from_params`?

Reply:
At the end of `load_archive`, `Model.load` is called, which calls `model_class._load` in `model.py`, which calls `Model.from_params`, etc. At some point it leads right back to `load_archive`.

Here's the sequence of calls I see when I was debugging via `pudb`:
```
load_archive archival.py:183
load model.py:396
_load model.py:296
from_params from_params.py:595
from_params from_params.py:628
from_archive model.py:440
load_archive archival.py:160
```

In order to ensure that the second call to `load_archive` also resulted in the same behavior, it was necessary to pass the argument through. I lack the details about why this would be necessary, however.

Reply:
Seeing this comment raised a flag for me that made me look into it.  There's no issue here, though.  This happened because the model type was "from_archive".  So, we first try to load the initial archive (the first call to `load_archive`).  When we're loading that archive, we see a path to another archive file specified (inside of the model params), which is where the nested call to `load_archive` happens.

But, I'm glad that there's a better solution that removes the need for this line.